### PR TITLE
Add experimental code to remove needles and caps using flips and collapses

### PR DESCRIPTION
**Pull request by sloriot**
The code is not public as a last step that requires the walker must be done before the function is removing all almost degenerate elements (but the function still improve the situation).

## Release Mangement
Affected package: `Polygon_mesh_processing`
Fixes: #3622

### CODE REVIEW

**Comment by sloriot** [comment]
@maxGimeno Could you please add an action under the repair/exp in the demo with a dialog box with 3 parameters?

**Comment by maxGimeno** [comment]
Careful, travis indicates a problem with licenses.

File: Polygon_mesh_processing/include/CGAL/polygon_mesh_processing.h [hunk 1]
```diff
@@ -54,6 +54,7 @@
 #include <CGAL/Polygon_mesh_processing/merge_border_vertices.h>
 #include <CGAL/Polygon_mesh_processing/smooth_mesh.h>
 #include <CGAL/Polygon_mesh_processing/smooth_shape.h>
+#include <CGAL/Polygon_mesh_processing/remove_degeneracies.h>
```
Comment (maxGimeno) [inline Polygon_mesh_processing/include/CGAL/polygon_mesh_processing.h]: There is an error here : this header is in  internal/

Comment (lrineau) [reply Polygon_mesh_processing/include/CGAL/polygon_mesh_processing.h]: This header `<CGAL/polygon_mesh_processing.h>` should be tested in the testsuite, or removed.

Comment (sloriot) [reply Polygon_mesh_processing/include/CGAL/polygon_mesh_processing.h]: It should have been catched by travis, it is a documented header. @maxGimeno how did you catch it?

Comment (maxGimeno) [reply Polygon_mesh_processing/include/CGAL/polygon_mesh_processing.h]: The swig bindings for CGAL master did not compile because of this.

Comment (lrineau) [reply Polygon_mesh_processing/include/CGAL/polygon_mesh_processing.h]: It is indeed documented: https://cgal.geometryfactory.com/CGAL/doc/master/Polygon_mesh_processing/polygon__mesh__processing_8h.html

Comment (lrineau) [reply Polygon_mesh_processing/include/CGAL/polygon_mesh_processing.h]: Actually, our script does not check headers that have a documentation page, but only headers that a referenced in other documentation pages.

Comment (lrineau) [reply Polygon_mesh_processing/include/CGAL/polygon_mesh_processing.h]: I have a patch:
```patch
diff --git a/Installation/CMakeLists.txt b/Installation/CMakeLists.txt
index c149c941c4e..db6e3120e77 100644
--- a/Installation/CMakeLists.txt
+++ b/Installation/CMakeLists.txt
@@ -1191,7 +1191,7 @@ You must disable CGAL_ENABLE_CHECK_HEADERS.")
   file(GLOB html_files RELATIVE "${DOC_DIR}/doc_output/" "${DOC_DIR}/doc_output/*/*.html")
   file(GLOB example_files RELATIVE "${CMAKE_SOURCE_DIR}/" "${CMAKE_SOURCE_DIR}/*/examples/*/*.cpp")
   find_program(AWK awk)
-  set(awk_arguments [=[{ match($0, /# *include *(&lt;|[<"])(CGAL\/[^>&"]*)([>"]|&gt;)/,arr); if(arr[2]!="") print arr[2] }]=])
+  set(awk_arguments [=[{ match($0, /# *include *(&lt;|[<"])(CGAL\/[^>&"]*)([>"]|&gt;)| (CGAL\/[^>&"]*\.h)/,arr); if(arr[2]!="") print arr[2]; if(arr[4]!="") print arr[4] }]=])
   message("listing headers from html files")
   foreach(f ${html_files})
     execute_process(COMMAND "${AWK}" "${awk_arguments}" "${DOC_DIR}/doc_output/${f}"
```

Comment (lrineau) [reply Polygon_mesh_processing/include/CGAL/polygon_mesh_processing.h]: That patch will add the following files to the list of documented files:
- `CGAL/boost/graph/partition.h`
- `CGAL/double.h`
- `CGAL/float.h`
- `CGAL/int.h`
- `CGAL/long_double.h`
- `CGAL/long_long.h`
- `CGAL/Mesh_3/initialize_triangulation_from_labeled_image.h`
- `CGAL/number_type_config.h`
- `CGAL/polygon_mesh_processing.h`
- `CGAL/simplest_rational_in_interval.h`
- `CGAL/subdivision_method_3.h`
- `CGAL/Surface_mesh_shortest_path.h`
- `CGAL/to_rational.h`
- `CGAL/utils.h`

**Pull request merged**